### PR TITLE
Feature 43 code server search

### LINKED ISSUE #43: Code server search
As a user I would like to be able to use code search to fill form fields automatically

### PR DESCRIPTION
**Pull request by Chi-Chu319**
resolves #43

### CODE REVIEW

**Review by EVP1989 — COMMENTED** [review]
Looks good

File: src/MetaformAutocompleteField.tsx [hunk 1]
```diff
@@ -47,113 +35,20 @@ export class MetaformAutocompleteFieldComponent extends React.Component<Props, S
...
-      }
-    });
+    const {
+      field,
+      formReadOnly,
+      value,
+      renderAutocomplete
+    } = this.props;
+
+    return renderAutocomplete(field, formReadOnly, value);
```
Comment (anttileppa) [inline src/MetaformAutocompleteField.tsx:52]: Wouldn't it be easier to pass just readOnly property, that combines formReadOnly and field.readyonly properties into one readonly value?

File: src/MetaformAutocompleteField.tsx [hunk 2]
```diff
@@ -1,36 +1,24 @@
...
-  value: FieldValue,
-  onValueChange: (value: FieldValue) => void,
-  setAutocompleteOptions: (path: string, input?: string) => Promise<string[] | MetaformAutocompleteItem[]>,
-  onFocus: () => void
+  field: MetaformField;
+  fieldId: string;
+  fieldLabelId: string;
+  formReadOnly: boolean;
+  value: FieldValue;
+  renderAutocomplete: (field: MetaformField, formReadOnly: boolean, value: FieldValue) => JSX.Element;
```
Comment (anttileppa) [inline src/MetaformAutocompleteField.tsx:14]:
```suggestion
  renderAutocomplete: (field: MetaformField, readOnly: boolean, value: FieldValue) => JSX.Element;
```

Comment (anttileppa) [reply src/MetaformAutocompleteField.tsx:14]: See comment on render method

File: src/MetaformComponent.tsx [hunk 3]
```diff
@@ -23,8 +22,8 @@ interface Props {
   setFieldValue: (fieldName: string, fieldValue: FieldValue) => void;
   datePicker: (fieldName: string, onChange: (date: Date) => void) => JSX.Element;
   datetimePicker: (fieldName: string, onChange: (date: Date) => void) => JSX.Element;
+  renderAutocomplete: (field: MetaformField, formReadOnly: boolean, value: FieldValue) => JSX.Element;
```
Comment (anttileppa) [inline src/MetaformComponent.tsx:25]:
```suggestion
  renderAutocomplete: (field: MetaformField, readOnly: boolean, value: FieldValue) => JSX.Element;
```

File: src/MetaformFieldComponent.tsx [hunk 4]
```diff
@@ -39,12 +39,12 @@ interface Props {
   setFieldValue: (fieldName: string, fieldValue: FieldValue) => void;
   datePicker: (fieldName: string, onChange: (date: Date) => void) => JSX.Element;
   datetimePicker: (fieldName: string, onChange: (date: Date) => void) => JSX.Element;
+  renderAutocomplete: (field: MetaformField, formReadOnly: boolean, value: FieldValue) => JSX.Element;
```
Comment (anttileppa) [inline src/MetaformFieldComponent.tsx:42]:
```suggestion
  renderAutocomplete: (field: MetaformField, readOnly: boolean, value: FieldValue) => JSX.Element;
```

File: src/MetaformSectionComponent.tsx [hunk 5]
```diff
@@ -23,8 +22,8 @@ interface Props {
   setFieldValue: (fieldName: string, fieldValue: FieldValue) => void;
   datePicker: (fieldName: string, onChange: (date: Date) => void) => JSX.Element;
   datetimePicker: (fieldName: string, onChange: (date: Date) => void) => JSX.Element;
+  renderAutocomplete: (field: MetaformField, formReadOnly: boolean, value: FieldValue) => JSX.Element;
```
Comment (anttileppa) [inline src/MetaformSectionComponent.tsx:25]:
```suggestion
  renderAutocomplete: (field: MetaformField, readOnly: boolean, value: FieldValue) => JSX.Element;
```

**Review by anttileppa — CHANGES_REQUESTED** [review]

**Pull request merged**